rmw_zenoh_cpp/src/detail/zenoh_router_check.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "zenoh_router_check.hpp"
+
+#include <rcutils/env.h>
+#include <rcutils/logging_macros.h>
+
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+namespace
+{
+
+// Convert a Zenoh Id to a string
+// Zenoh IDs are LSB-first 128bit unsigned and non-zero integers in hexadecimal lowercase.
+// @param pid Zenoh Id to convert
+std::string ZidToStr(z_id_t pid)
+{
+  std::stringstream ss;
+  int len = 0;
+  for (int i = 0; i < 16; i++) {
+    if (pid.id[i]) {
+      len = i + 1;
+    }
+  }
+  if (!len) {
+    ss << "";
+  } else {
+    for (int i = len - 1; i >= 0; --i) {
+      ss << std::hex << std::setfill('0') << std::setw(2) << static_cast<int>(pid.id[i]);
+    }
+  }
+  return ss.str();
+}
+
+}  // namespace
+
+rmw_ret_t zenoh_router_check(z_session_t session)
+{
+  // Initialize context for callback
+  void * context = malloc(sizeof(int));
+  *(static_cast<int *>(context)) = 0;
```

Comment:
Hm, does this need to be heap-allocated memory?  Could this just be a stack-allocated `int` that we capture in the callback?

Reply:
Thanks for the suggestion. Yeah it wasn't necessary.

Done.